akka-serialization-jackson/src/main/scala/akka/serialization/jackson/JacksonSerializer.scala

```diff
@@ -220,9 +263,9 @@ import com.fasterxml.jackson.dataformat.cbor.CBORFactory
     val startTime = if (isDebugEnabled) System.nanoTime else 0L
 
     val (fromVersion, manifestClassName) = parseManifest(manifest)
-    checkAllowedClassName(manifestClassName)
+    if (typeInManifest) checkAllowedClassName(manifestClassName)
 
-    val migration = migrations.get(manifestClassName)
+    val migration = migrations.get(deserializationType.fold(manifestClassName)(_.getName))
```

Comment:
```suggestion
    val migration = migrations.get(deserializationType.map(_.getName).getOrElse(manifestClassName))
```

Reply:
While I agree that `fold` on `Optional` is difficult to get used to I don't think it's worth holding back merging this. Please follow up @raboof if you like.